server/app/views/api/ApiDocsView.java

```diff
@@ -137,17 +138,7 @@ private DivTag contentDiv(
 
   private DivTag apiResponseSampleDiv(ProgramDefinition programDefinition) {
     DivTag apiResponseSampleDiv = div();
-
-    ImmutableList<QuestionDefinition> questionDefinitions =
-        programDefinition.streamQuestionDefinitions().collect(toImmutableList());
-
-    CfJsonDocumentContext sampleJson = new CfJsonDocumentContext();
-    for (QuestionDefinition questionDefinition : questionDefinitions) {
-      sampleJson.mergeFrom(
-          questionJsonSamplerFactory
-              .create(questionDefinition.getQuestionType())
-              .getSampleJson(questionDefinition));
-    }
+    CfJsonDocumentContext sampleJson = programJsonSampler.getSampleJson(programDefinition);
```

Comment:
It shouldn't have been the responsibility of the view to merge `CfJsonDocumentContext`s, so now we delegate it to the `ProgramJsonSampler`.